tests/cli/workflow/rnafusion/test_cli_rnafusion_config_case.py

```diff
@@ -52,110 +23,141 @@ def test_without_samples(
-        assert option_value in caplog.text
+    caplog.set_level(logging.ERROR)
 
+    # GIVEN a VALID case_id and non-accepted strandedness option
 
-def test_wrong_strandedness(
-    cli_runner: CliRunner,
-    rnafusion_context: CGConfig,
-    caplog: LogCaptureFixture,
-    rnafusion_case_id: str,
-):
-    """Test command with --strandedness option."""
-    caplog.set_level(logging.INFO)
-    # GIVEN a VALID case_id and genome_version
-    case_id: str = rnafusion_case_id
-    option_key: str = "--strandedness"
-    option_value: str = "unknown"
     # WHEN running with strandedness option specified
     result = cli_runner.invoke(
         config_case,
-        [case_id, option_key, option_value],
+        [rnafusion_case_id, "--strandedness", rnafusion_strandedness_not_acceptable],
         obj=rnafusion_context,
     )
+
     # THEN command should fail
     assert result.exit_code != EXIT_SUCCESS
+    assert "Could not create config files for" in caplog.text
+    assert "validation error" in caplog.text
 
 
-def test_params_file(
+def test_defaults(
     cli_runner: CliRunner,
     rnafusion_context: CGConfig,
-    caplog: LogCaptureFixture,
     rnafusion_case_id: str,
+    rnafusion_sample_sheet_path: Path,
+    rnafusion_params_file_path: Path,
+    rnafusion_sample_sheet_content: str,
+    caplog: LogCaptureFixture,
 ):
-    """Test that command generates default params_file."""
+    """Test that command generates default config files."""
     caplog.set_level(logging.INFO)
 
-    # GIVEN a VALID case_id and genome_version
-    case_id: str = rnafusion_case_id
+    # GIVEN a valid case
 
     # WHEN running config case
-    result = cli_runner.invoke(config_case, [case_id], obj=rnafusion_context)
+    result = cli_runner.invoke(config_case, [rnafusion_case_id], obj=rnafusion_context)
 
-    # THEN command should print the rnafusion command-string
+    # THEN command should exit succesfully
     assert result.exit_code == EXIT_SUCCESS
 
+    # THEN sample sheet file should be generated
+    assert "Getting sample sheet information" in caplog.text
+    assert "Writing sample sheet" in caplog.text
+
     # THEN parameters file should be generated
-    assert "Generating parameters file" in caplog.text
+    assert "Getting parameters information" in caplog.text
+    assert "Writing parameters file" in caplog.text
+    assert rnafusion_sample_sheet_path.is_file()
+    assert rnafusion_params_file_path.is_file()
+
+    # THEN the sample sheet content should match the expected values
+    with rnafusion_sample_sheet_path.open("r") as file:
+        content = file.read()
+        assert ",".join(RnafusionSample.headers()) in content
+        assert rnafusion_sample_sheet_content in content
+
+    # THEN the params file should contain all parameters
+    with rnafusion_params_file_path.open("r") as file:
+        content = file.read()
+        for parameter in RnafusionParameters.__annotations__.keys():
```

Comment:
Is there a way to do this without using private class methods, i.e. (__). It is not best practice and should be avoided even in tests.

Reply:
not that I know of unless making a list with all parameters

Reply:
If you make a  RnafusionParameters test fixture initialized with the defaults and supply the fixture here.

Use: `vars(rnafusion_parameters_default).keys()` to get the model attributes and use them in the assert.

That will make it clear what the defaults should be without defining them in the test and using private methods

Reply:
done